### ISSUE TITLE
P2: NVIDIA MIG parallel ensemble inference for Analysis Flow

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/minivess-mlops#883]
## Context

The Analysis Flow currently evaluates ensemble members sequentially. With NVIDIA Multi-Instance GPU (MIG) or multi-GPU setups, members could be evaluated in parallel.

### MIG detection exists:
\`src/minivess/serving/mig_config.py\` provides:
- \`MIGConfig\` dataclass — configuration for MIG-based multi-model serving (\`gpu_index\`, \`instances\`)
- \`detect_mig_devices()\` — detects MIG-partitioned GPUs using \`pynvml\`, returns list of \`{gpu_index, instance_id, memory_gb}\`
- \`assign_models_to_instances()\` — round-robin assignment of models to MIG instances

### Sequential ensemble inference:
\`src/minivess/ensemble/strategies.py\` \`EnsemblePredictor.predict()\` loops through members sequentially:
\`\`\`python
predictions: list[Tensor] = []
for model in self.models:
    model.eval()
    output = model(images, **kwargs)
    predictions.append(output.prediction)
stacked = torch.stack(predictions, dim=0)
\`\`\`

### Existing UQ infrastructure:
- \`src/minivess/ensemble/deep_ensembles.py\` — \`DeepEnsemblePredictor\` with uncertainty decomposition (also sequential)
- \`src/minivess/ensemble/mc_dropout.py\` — \`MCDropoutPredictor\` (sequential forward passes)
- \`src/minivess/ensemble/builder.py\` — \`EnsembleMember\` dataclass, 4 ensemble building strategies

## What's Missing

1. **MIG-aware inference dispatcher** — route ensemble members to specific MIG instances based on \`assign_models_to_instances()\` output
2. **\`torch.cuda.set_device()\` integration** — set the correct MIG device before each member's forward pass
3. **Process-level parallelism** — for non-MIG multi-GPU, use \`torch.multiprocessing\` or \`concurrent.futures.ProcessPoolExecutor\` to run members on different GPUs
4. **Batch dimension trick** — for same-architecture ensembles, stack inputs along batch dimension and run a single forward pass (most efficient)
5. **Fallback to sequential** — when MIG is not available or only 1 GPU, fall back to current sequential evaluation
6. **Integration with \`EnsemblePredictor\`** — the parallel dispatcher should be transparent to the strategy layer

## Relevant Files

- \`src/minivess/serving/mig_config.py\` — \`MIGConfig\`, \`detect_mig_devices()\`, \`assign_models_to_instances()\` (130 lines, complete)
- \`src/minivess/ensemble/strategies.py\` — \`EnsemblePredictor\` with sequential predict loop (148 lines)
- \`src/minivess/ensemble/deep_ensembles.py\` — \`DeepEnsemblePredictor\` with uncertainty decomposition
- \`src/minivess/ensemble/builder.py\` — \`EnsembleMember\`, \`EnsembleSpec\`, 4 build strategies

## Blockers

- MIG requires A100/A30/H100 — not available on current dev hardware (RTX 2070 Super) or RunPod RTX 4090
- Testing requires access to a MIG-capable GPU (GCP A100 spot)
- Not blocking any current workflow — pure performance optimization

## Priority Context

Current sequential evaluation works correctly. This is a performance optimization for large ensembles (e.g., \`all_loss_all_best\` with 54+ members from the full factorial design: 6 models x 3 folds x 6 metric checkpoints = 108 members max). For the paper deadline, sequential evaluation is sufficient. MIG parallelism matters for production inference where latency budgets are tight.

## References

- [NVIDIA MIG documentation](https://www.nvidia.com/en-us/technologies/multi-instance-gpu/)
- User request from 2026-03-20 session
